src/lyapunovspectrum.jl

```diff
@@ -131,44 +130,76 @@ function liap_jetcoeffs!(eqsdiff!, t::Taylor1{T}, x::Vector{Taylor1{U}},
     nothing
 end
 
-"""
-    liap_taylorstep!(f, x, dx, xaux, δx, dδx, jac, t0, t1, x0, order, abstol, vT)
-
-Similar to [`taylorstep!`](@ref) for the calculation of the Liapunov
-spectrum. `jac` is the linearization of the equations of motion,
-and `xaux`, `δx`, `dδx` and `vT` are auxiliary vectors.
+function variational_eqs!(t::Taylor1{T}, x::Vector{Taylor1{U}},
+        dx::Vector{Taylor1{U}}, jac::Array{Taylor1{U},2}, eqsdiff_jac! =Nothing) where {T<:Real, U<:Number}
+    # Dimensions of phase-space: dof
+    nx = length(x)
+    dof = size(jac, 1)
+    if eqsdiff_jac! != Nothing
+        # Stability matrix
+        eqsdiff_jac!(jac, t, x)
+    end
+    @inbounds dx[dof+1:nx] = jac * reshape( x[dof+1:nx], (dof,dof) )
+    return nothing
+end
 
 """
-function liap_taylorstep!(f, t::Taylor1{T}, x::Vector{Taylor1{U}}, dx::Vector{Taylor1{U}},
-        xaux::Vector{Taylor1{U}}, δx::Array{TaylorN{Taylor1{U}},1},
-        dδx::Array{TaylorN{Taylor1{U}},1}, jac::Array{Taylor1{U},2}, t0::T, t1::T, x0::Array{U,1},
-        order::Int, abstol::T, _δv::Array{TaylorN{Taylor1{U}}}) where {T<:Real, U<:Number}
+    lyap_taylorstep!(f!, t, x, dx, xaux, δx, dδx, jac, t0, t1, x0, order, abstol, _δv[, f])
 
-    # Compute the Taylor coefficients
-    liap_jetcoeffs!(f, t, x, dx, xaux, δx, dδx, jac, _δv)
+Similar to [`taylorstep!`](@ref) for the calculation of the Lyapunov spectrum.
+`jac` is the current value of the linearization of the equations of motion, i.e,
+the Jacobian. `xaux`, `δx`, `dδx` and `vT` are auxiliary vectors. Optionally, the
+user may provide an Jacobian function `f!` to evaluate the current value of the
+Jacobian. For more details on `f!`, see [`lyap_jetcoeffs!`](@ref).
 
+"""
+function lyap_taylorstep!(f!, t::Taylor1{T}, x::Vector{Taylor1{U}},
+        dx::Vector{Taylor1{U}}, xaux::Vector{Taylor1{U}},
+        δx::Array{TaylorN{Taylor1{U}},1}, dδx::Array{TaylorN{Taylor1{U}},1},
+        jac::Array{Taylor1{U},2}, t0::T, t1::T, x0::Array{U,1}, order::Int,
+        abstol::T, _δv::Array{TaylorN{Taylor1{U}}}, jac! =Nothing) where {T<:Real, U<:Number}
+    # Dimensions of phase-space: dof
+    nx = length(x)
+    dof = length(δx)
+    jetcoeffs!(f!, t, view(x, 1:dof), view(dx, 1:dof), view(xaux, 1:dof))
+    if jac! == Nothing
+        # lyap_jetcoeffs!(f!, t, x, dx, xaux, δx, dδx, jac, _δv)
```

Comment:
I think the contents of the first part of the `if` (at least partially) could be moved to a function. I am not sure if that would be better for performance, but it would allow testing the user-provided function for the Jacobian against AD, and that's time saving!